src/WebJobs.Script/Host/ScriptHostManager.cs

```diff
@@ -340,6 +340,13 @@ private ScriptHost[] GetLiveInstancesAndClear()
 
         protected virtual void OnInitializeConfig(ScriptHostConfiguration config)
         {
+            var loggingConnectionString = config.HostConfig.DashboardConnectionString;
+            if (string.IsNullOrEmpty(loggingConnectionString))
+            {
+                // if no Dashboard connection string is provided, set this to null
+                // to prevent host startup failure
+                config.HostConfig.DashboardConnectionString = null;
```

Comment:
In core SDK, if they don't want any Dashboard logging, we require them to explicitly set this to null. Otherwise the host will throw. In our WebHost we always set it to null (see above), but for the console Host, we want to set to null IFF no DB connection string is present.